MatrixSDK/Data/RoomList/MXRoomListDataFilterOptions.swift

```diff
@@ -121,6 +121,11 @@ public final class MXRoomListDataFilterOptions: NSObject {
         }
         
         if !onlySuggested {
+            // Filter out any room previews by making sure the user has valid membership.
+            let memberPredicate = NSPredicate(format: "%K != %d",
```

Comment:
I would prefer to make this configurable with a property like `hideUnknownMembershipRooms` with a default value of `true`, as i don't like the idea of having such logic in this class.